letter-shop/index.md

```diff
@@ -0,0 +1,634 @@
+
+:tada: Congratulations, that's step 4 of 5 complete for our server!
+
+But, our client is still waiting for the fulfillment. Let's go back to the client and finish this up!
+
+## Step 5: Using the fulfillment to get the Letter
+
+As you've proabably guessed, we're missing an event listener in our client too. Go back to `pay.js` and replace `// Listen for fulfillments...` with the following code:
+
+```js
+  //Send the transfer
+  plugin.on('outgoing_fulfill', function (transferId, fulfillment) {
+    console.log(`    - Transfer executed. Got fulfillment: ${fulfillment}`)
+    console.log(` 3. Collect your letter at http://localhost:8000/${fulfillment}`)
+    plugin.disconnect()
+    process.exit()
+  })
+```
+
+As the transfer initiator we are interested in fulfillments that come out of the ledger after being submitted by the receiver, so we listen for the `outgoing_fulfill` event.
+
+All we need to do with that fulfillment in this instance is print it to the screen so the user can go and collect their letter.
+
+Re-run your client and after a few seconds you should now get a response to that effect. 
+
+:tada: Congratulations! You've now finished the client.
+
+So all that's left to do is get that letter! Go back to your browser and paste the URL that was output by the client into your address bar and you should receive the letter that was prepared for you upon your first request.
+
+If you switch back to the terminal window for your server you'll see some additional output showing that the shop server accepted the request, looked up the letter using the fulfillment, and returned this letter.
+
+:tada: Congratulations, that's step 5 of 5 and you now have a working server too.
+
+## Bonus Step: A Paying Proxy
+
+It's very cumbersome to copy and paste the condition from your browser to your command-line terminal each time you need to pay for something online, and then to copy and paste back the fulfillment from your terminal to your browser once you have paid. 
+
+For most paid services we'd expect the client to detect that a payment is required, make the payment, and then retry the request.
+
+To do this we need to make a few tweaks to the server and client.
+
+First, we need to make our responses machine readable so that the client can find the details of the payment required in the response. Recall how we used the 402 (Payment Required) HTTP response in the server when we responded to the initial request. Now we can see how this is useful for clients to help them recognise the response type.
+
+Another minor change that required is to put the payment details in a response header in a standard format that clients recognise.
+
+Modify the code in `shop.js` to look like this:
+
+```js
+      console.log(`    - Waiting for payment...`)
+
+      res.setHeader("Pay", `${cost} ${account} ${condition}`)
```

Comment:
In http-ilp the third string would be the PSK shared secret, not the one-off condition.

Reply:
That's okay, we're still building to that. That's why I haven't said anything about HTTP ILP just showing what's possible